core/app/[locale]/(default)/product/[slug]/_actions/add-to-cart.tsx

```diff
@@ -61,13 +83,15 @@ export async function handleAddToCart(
 
         return { ...accum, multipleChoices: [multipleChoicesOptionInput] };
 
-      case 'CheckboxOption':
+      case 'checkbox':
         checkboxOptionInput = {
-          optionEntityId: option.entityId,
+          optionEntityId: Number(field.name),
           optionValueEntityId:
-            Number(optionValueEntityId) !== 0
-              ? option.checkedOptionValueEntityId
-              : option.uncheckedOptionValueEntityId,
+            optionValueEntityId === 'true'
+              ? // @ts-expect-error Need to fix types
+                Number(field.checkedValue)
+              : // @ts-expect-error Need to fix types
+                Number(field.uncheckedValue),
```

Comment:
Do we need to fix these right now?

Reply:
I forgot about these, let me take a look 👀

Reply:
Other than updating the schema which I think @migueloller was hesitant to include because this is a very BC pattern:

```ts
type CheckboxField = {
  type: 'checkbox';
  defaultValue?: string;
  checkedValue?: string;
  uncheckedValue?: string
} & FormField;
```

I can solve by passing the "unparsed" product options to my action:

```ts
// Action I pass to ProductDetails
const action = async (prevState: State, payload: FormData) => {
  'use server';

  const product = await productPromise;

  return await addToCart(prevState, payload, product.productOptions);
};
  
// In action function
case 'checkbox':
// eslint-disable-next-line no-case-declarations <-- still need this or I can move outside of the switch I guess.
const checkbox = checkboxOptions.find((option) => option.entityId === Number(field.name));

if (!checkbox) {
  return { ...accum };
}

checkboxOptionInput = {
  optionEntityId: Number(field.name),
  optionValueEntityId:
    optionValueEntityId === 'true'
      ? checkbox.checkedOptionValueEntityId
      : checkbox.uncheckedOptionValueEntityId,
};
....
```

What do you think?

Reply:
The approach we had decided to take on this was to allow fields to be generic so you could pass additional fields that aren't necessarily used by VIBES and those would be passed back to the server action. Perhaps we also need to introduce the ability to pass the Zod schema for the form or we can simply make the Zod schema in VIBES use `passthrough` and in the server action, the VIBES schema can be extended to validate those fields are being added.

```ts
async function action() {
  const submission = parseWithZod(formData, { schema: schema.extend({ /* extend schema here */}) })
}
```